lib/install/validate/validator.go

```diff
@@ -357,6 +359,93 @@ func (v *Validator) generateBridgeName(ctx, input *data.Data, conf *metadata.Vir
 	return input.DisplayName
 }
 
+func (v *Validator) firewall(ctx context.Context) {
+	defer trace.End(trace.Begin(""))
+
+	var hosts []*object.HostSystem
+	var err error
+
+	rule := types.HostFirewallRule{
+		Port:      8080, // serialOverLANPort
+		PortType:  types.HostFirewallRulePortTypeDst,
+		Protocol:  string(types.HostFirewallRuleProtocolTcp),
+		Direction: types.HostFirewallRuleDirectionOutbound,
+	}
+
+	if hosts, err = v.Session.Datastore.AttachedClusterHosts(ctx, v.Session.Cluster); err != nil {
+		log.Errorf("Unable to get the list of hosts attached to given storage: %s", err)
+		v.NoteIssue(err)
+		return
+	}
+
+	var misconfiguredEnabled []string
+	var misconfiguredDisabled []string
+	var correct []string
+
+	for _, host := range hosts {
+		fs, err := host.ConfigManager().FirewallSystem(ctx)
+		if err != nil {
+			v.NoteIssue(err)
+			break
+		}
+
+		disabled := false
+		esxfw, err := esxcli.GetFirewallInfo(host)
+		if err != nil {
+			v.NoteIssue(err)
+			break
+		}
+		if !esxfw.Enabled {
+			disabled = true
+			log.Infof("Firewall status: DISABLED on %s", host.InventoryPath)
+		} else {
+			log.Infof("Firewall status: ENABLED on %s", host.InventoryPath)
+		}
+
+		info, err := fs.Info(ctx)
+		if err != nil {
+			v.NoteIssue(err)
+			break
+		}
+
+		rs := object.HostFirewallRulesetList(info.Ruleset)
+		_, err = rs.EnabledByRule(rule, true)
+		if err != nil {
+			if !disabled {
+				misconfiguredEnabled = append(misconfiguredEnabled, host.InventoryPath)
+			} else {
+				misconfiguredDisabled = append(misconfiguredDisabled, host.InventoryPath)
+			}
+		} else {
+			correct = append(correct, host.InventoryPath)
+		}
+	}
+
+	if len(correct) > 0 {
+		log.Info("Firewall configuration OK on hosts:")
+		for _, h := range correct {
+			log.Infof("  %s", h)
+		}
+	}
+	if len(misconfiguredEnabled) > 0 {
+		log.Error("Firewall configuration incorrect on hosts:")
+		for _, h := range misconfiguredEnabled {
+			log.Errorf("  %s", h)
+		}
+		msg := "Firewall must permit 8080/tcp outbound to use VIC"
+		log.Error(msg)
+		v.NoteIssue(errors.New(msg))
```

Comment:
Would suggest to noteIssue while no any host is correctly configured, instead of to have all of them enable this policy.
In the future, user might complain that I have some hosts works with VIC, but you requires all of them match your rules.